### PR TITLE
Tokenizer implemented

### PR DESCRIPTION
**Pull request by EugeneDar**
Closes: #15

---

## PR-Codex overview
This PR adds license headers and documentation to the codebase, as well as new token classes and a tokenizer test.

### Detailed summary
- Added MIT license header to all files
- Added package-info.java files with documentation
- Added Token, ArrowToken, StringToken, and BracketToken classes
- Added TokenizerTest class for testing the Tokenizer class
- Updated Tokenizer class to tokenize input string into tokens

> The following files were skipped due to too many changes: `src/main/java/org/objectionary/Tokenizer.java`, `pom.xml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by EugeneDar** [comment]
@Graur please, have a look

File: src/test/java/org/objectionary/TokenizerTest.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        MatcherAssert.assertThat(
+            (BracketToken) tokenizer.getToken(), Matchers.equalTo(BracketToken.BracketType.CLOSE)
+        );
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.getToken(), Matchers.instanceOf(BracketToken.class));
+        MatcherAssert.assertThat(
+            (BracketToken) tokenizer.getToken(), Matchers.equalTo(BracketToken.BracketType.CLOSE)
+        );
+        tokenizer.next();
+        MatcherAssert.assertThat(tokenizer.hasNext(), Matchers.equalTo(false));
```
Comment (Graur) [inline src/test/java/org/objectionary/TokenizerTest.java:78]: @EugeneDar Maybe this test can be implemented via `while` or `for` loop. What do you think?

File: src/main/java/org/objectionary/tokens/Token.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.objectionary.tokens;
+
+/**
+ * This class represents the token.
+ * @since 0.1.0
+ */
+public class Token {
```
[the remark this answers is not in the document]
Comment (EugeneDar) [reply src/main/java/org/objectionary/tokens/Token.java:30]: @Graur it has descendant classes, so no.

File: src/test/java/org/objectionary/TokenizerTest.java (hunk 1 above)
Comment (EugeneDar) [reply src/test/java/org/objectionary/TokenizerTest.java:78]: @Graur sure, I will fix it.

**Comment by EugeneDar** [comment]
@Graur please, have a look again

**Comment by EugeneDar** [comment]
@yegor256 please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Comment by EugeneDar** [comment]
@yegor256 Can you please clarify if it is something I need to fix or if it is a problem with the docker container

**Comment by yegor256** [comment]
@EugeneDar install xcop locally and check all XML files with it, you will see the problem

**Comment by EugeneDar** [comment]
@yegor256 please, merge again

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**